### ISSUE TITLE
Feature request: Memory only caching 

### ISSUE THREAD

**Issue by finBitorg** [requests-cache/requests-cache#846]
### Use case
On raspberrypi writing cache file to sdcards kills the sd card. Please add memory only switch to allow the cache to be stored in memory only and maybe with a max cache size parameter. 

for example : SESSION = CachedSession('demo_cache', expire_after=300, mem_only=0)

where mem_only=0 is mem only caching with no size limit; mem_only=1000000 is mem only caching with 1000000 bytes/ 1MB max ram usage for cache; mem_only=None/Not specified  then use file based cache as normal.

**Comment by JWCook** [comment]
Did you find what you needed? Some options are:
* A basic [in-memory cache](https://requests-cache.readthedocs.io/en/stable/user_guide/backends.html) (`backend='memory'`)
* A [SQLite in-memory database](https://requests-cache.readthedocs.io/en/stable/user_guide/backends/sqlite.html#in-memory-caching) (`backend=SQLiteCache(use_memory=True)`)
* If you wanted to, using one of the options above, you could manually read/write to storage when you choose (for example only on application startup/shutdown).

**Comment by finBitorg** [comment]
Yes this works; I didn't see it in the docs at first. Thanks for this awesome project !!

**Comment by finBitorg** [comment]
btw this might be documented somewhere but what's the benefit of SQL lite in memory vs basic ?

**Comment by JWCook** [comment]
There won't be a noticeable difference for most use cases, but for advanced or specialized usage, SQLite will give you some more features. For example:
* If you wanted to do any lower-level work with the underlying database, you'd have all the features of the python [sqlite3](https://docs.python.org/3/library/sqlite3.html) module.
* Some operations might be a bit faster in SQLite due to extra indexes, like clearing expired responses ([`session.cache.delete(expired=True)`](https://requests-cache.readthedocs.io/en/stable/modules/requests_cache.backends.base.html#requests_cache.backends.base.BaseCache.delete)). 
* For large caches, using [`backup()`](https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.backup) to copy an in-memory db to disk (or vice versa) would be faster than copying with [`session.cache.update()`](https://requests-cache.readthedocs.io/en/stable/user_guide/backends.html#exporting-to-a-different-backend).

It's possible that one of those backends might be more memory-efficient than the other in some situations, but I haven't benchmarked that in a long time. I'll leave this issue open as a reminder to do that.

**Comment by finBitorg** [comment]
Ok last question I promise; I see docs for custom filtering function that returns a boolean,  but is possible to specify a custom cache time using a custom function? For example of the response is less than a megabyte cache it for 1hour else if more than ache it for only 20 minutes ?

**Comment by JWCook** [comment]
There's not an option to provide a callback for setting expiration time, but here's an example that accomplishes the same thing:
```python
from requests_cache import CachedSession
from datetime import datetime, timedelta

session = CachedSession(expire_after=timedelta(hours=1))
r = session.get(url)

# Set a shorter expiration for responses over 1MB
if len(r.content) > 1000000:
    expires = datetime.utcnow() + timedelta(minutes=20)  # Requires an absolute datetime in UTC
    session.cache.save_response(r, expires=expires)
```

**Comment by finBitorg** [comment]
is there a way to see what the specific expiratino is for a url or conent ? 

like session.cache.get_ttl(url or reponse object ?)
